### PR TITLE
Refactored variables to make the event pipeline action group optional

### PR DESCRIPTION
**Pull request by QBY-MarkusMaring**
# Description

Previously the module required a configuration for an event pipeline service_uri and secret. This update makes it so that the config is all in one variable and can be omitted if no event pipeline connection is needed.

# Change overview (tick true):

- [x] This introduces backward incompatible changes
- [ ] This adds a new backward compatible Feature
- [ ] This fixes a Bug

# Version information: 

- Previous Version: `2.0.3`

- Next Version based on [Semantic Versioning](https://semver.org/#summary) (see above): `3.0.0`

# How Has This Been Tested?

- [x] Apply of all examples was successful

# Checklist:

- [x] I have run tests and documented them above
- [x] I have performed a self-review of my own code
- [ ] I have updated the documentation
- [x] I have updated the CHANGELOG

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -50,3 +39,46 @@ variable "automation_account" {
...
+  {
+    enabled       = Enable the action group if you want to send data to a monitoring service.
+    name          = Name of the alert webhook.
+    service_uri   = Link to the webhook receiver URL. Must contain the placeholder \"{{pipeline_key}}\" which will be replaced with the secret from var.event_pipeline_key.
+}
+  ```
+  DOC
+  default = {
+    enabled = false
+  }
```
Comment (QBY-ThimoLimpert) [inline variables.tf]: Defining description two times lookswrong

File: variables.tf [hunk 2]
```diff
@@ -50,3 +39,46 @@ variable "automation_account" {
...
+}
+  ```
+  DOC
+
+  description = <<-DOC
+  ```
+  {
+    enabled       = Enable the action group if you want to send data to a monitoring service.
+    name          = Name of the alert webhook.
+    service_uri   = Link to the webhook receiver URL. Must contain the placeholder \"{{pipeline_key}}\" which will be replaced with the secret from var.event_pipeline_key.
```
Comment (QBY-ThimoLimpert) [inline variables.tf:55]: I don't get the explanation

File: variables.tf [hunk 3]
```diff
@@ -50,3 +39,46 @@ variable "automation_account" {
   description = "Automation account where the resource graph script will be deployed."
   nullable    = false
 }
+
+variable "event_pipeline_config" {
+  type = object({
+    enabled     = bool
+    name        = optional(string)
```
Comment (QBY-ThimoLimpert) [inline variables.tf:46]: is it reasonable to set a default?

File: main.tf [hunk 4]
```diff
@@ -34,13 +34,14 @@ module "vm_insights" {
...
+  count = var.event_pipeline_config.enabled ? 1 : 0
   name                = "EventPipelineCentral_AG_1"
   resource_group_name = var.log_analytics_workspace.resource_group_name
   short_name          = "monitorhook"
 
   webhook_receiver {
-    name                    = var.webhook_name
-    service_uri             = var.webhook_service_uri
+    name                    = var.event_pipeline_config.name
+    service_uri             = replace(var.event_pipeline_config.service_uri, "{{pipeline_key}}", var.event_pipeline_key)
```
Comment (QBY-ThimoLimpert) [inline main.tf:44]: why use magic instead of building the uri?

**Review by QBY-ThimoLimpert — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 1 above)
Comment (QBY-MarkusMaring) [reply variables.tf]: Copy paste mistake. We noticed that too :)

File: variables.tf [hunk 5]
```diff
@@ -50,3 +39,34 @@ variable "automation_account" {
...
+    name        = optional(string, "QBY EventPipeline")
+    service_uri = optional(string)
+  })
+
+  description = <<-DOC
+  ```
+  {
+    enabled       = Enable the action group if you want to send data to a monitoring service.
+    name          = Name of the alert webhook.
+    service_uri   = Link to the webhook receiver URL. Must contain the placeholder \"{{secret}}\". This placeholder will be replaced by the secret value from var.secret. This is used to add authentication to the webhook URL as a query parameter.
```
Comment (QBY-ThimoLimpert) [inline variables.tf:55]: What happens if it doesn't contain `{{secret}}`?

**Review by QBY-ThimoLimpert — APPROVED** [review]

File: variables.tf (hunk 5 above)
Comment (QBY-MarkusMaring) [reply variables.tf:55]: The validation in line 68 fails. It is assumed that the webhook is secured with a secret so the URI must contain the placeholder.

**Review by QBY-ThimoLimpert — APPROVED** [review]

**Pull request merged**